tests/explainers/test_tree.py

```diff
@@ -771,6 +772,57 @@ def test_xgboost_buffer_strip():
     assert isinstance(explainer, shap.explainers.Tree)
 
 
+class TestSingleTree:
+    """Tests for the SingleTree class."""
+
+    def test_singletree_lightgbm_basic(self):
+        """A basic test for checking that a LightGBM dump_model() dictionary
+        is parsed properly into a SingleTree object.
+        """
+
+        # Stump (only root node) tree
+        # FIXME: this test should NOT throw a KeyError, see #2044
+        with pytest.raises(KeyError, match="leaf_index"):
```

Comment:
`xfail` is a good shout. will keep this in mind for future. but since I've already implemented the fix and the fix-PR will be raised soon, I think we can let this slide :p